CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc

```diff
@@ -21,243 +22,85 @@ CaloTPGTranscoderULUT::CaloTPGTranscoderULUT(const std::string& compressionFile,
                                                   compressionFile_(compressionFile),
                                                   decompressionFile_(decompressionFile)
 {
-  for (int i = 0; i < NOUTLUTS; i++) outputLUT_[i] = 0;
+  outputLUT_.clear();
 }
 
 CaloTPGTranscoderULUT::~CaloTPGTranscoderULUT() {
-  for (int i = 0; i < NOUTLUTS; i++) {
-    if (outputLUT_[i] != 0) delete [] outputLUT_[i];
-  }
 }
 
 void CaloTPGTranscoderULUT::loadHCALCompress(HcalLutMetadata const& lutMetadata,
                                              HcalTrigTowerGeometry const& theTrigTowerGeometry) {
-// Initialize analytical compression LUT's here
-   // TODO cms::Error log
-  if (OUTPUT_LUT_SIZE != (unsigned int) 0x400) std::cout << "Error: Analytic compression expects 10-bit LUT; found LUT with " << OUTPUT_LUT_SIZE << " entries instead" << std::endl;
+    // Initialize analytical compression LUT's here
+    if (OUTPUT_LUT_SIZE != (unsigned int) 0x400)
+        edm::LogError("CaloTPGTranscoderULUT") << "Analytic compression expects 10-bit LUT; found LUT with " << OUTPUT_LUT_SIZE << " entries instead";
 
-  std::vector<unsigned int> analyticalLUT(OUTPUT_LUT_SIZE, 0);
-  std::vector<unsigned int> identityLUT(OUTPUT_LUT_SIZE, 0);
+    std::vector<unsigned int> analyticalLUT(OUTPUT_LUT_SIZE, 0);
+    std::vector<unsigned int> identityLUT(OUTPUT_LUT_SIZE, 0);
```

Comment:
if these have a fixed, known size, make them std::arrays. Especially good for both bounds checking and if you happen to be putting this sort of construct into a map